File: week-04/day-03/sharpie.py
from random import random

class Sharpie(object):
    def __init__(self, color, width, ink_amount = 100.0):
        self.color = color
        self.width = width
        self.ink_amount = ink_amount

    def use(self):
        self.ink_amount *= random()
        return self.ink_amount
    

# pen = Sharpie("green", 3.5)

# print("The pen color is {}, width is {} and the amount of the ink is {}.".format(
#     pen.color, pen.width, pen.ink_amount
# ))

# pen.use()

# print("The pen color is {}, width is {} and the amount of the ink is {}.".format(
#     pen.color, pen.width, pen.ink_amount
# ))

File: week-08/day-1/sharpie.js
'use strict';

// Create a Sharpie constructor function
// We should know about each sharpie:
// color (which should be a string)
// width (which will be a number)
// inkAmount (another number)
// When instantiating a Sharpie, we need to specify the color and the width
// Every sharpie is created with a default 100 as inkAmount
// We can use() the sharpie objects
// which decreases inkAmount by the width
// Write a loop that consumes all the sharpie's ink.
// Make sure your loop works with any inkAmount, so your code figures out when it's out of ink.

function Sharpie(color, width) {
  this.color = color,
  this.width = width,
  this.inkAmount = 100,
  this.use = function() {
    this.inkAmount -= this.width;
  }
}

let redPen = new Sharpie('red', 3.3);

for (let i = 1; redPen.inkAmount > 0; i++ ) {
  redPen.use();
  console.log(i);
}